apiserver/Dockerfile

```diff
@@ -5,16 +5,22 @@ WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY apiserver/go.mod apiserver/go.mod
 COPY apiserver/go.sum apiserver/go.sum
+COPY proto/go.mod proto/go.mod
+COPY proto/go.sum proto/go.sum
+COPY ray-operator/go.mod ray-operator/go.mod
+COPY ray-operator/go.sum ray-operator/go.sum
+
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
```

Comment:
I feel like it's a recommended practice, to download all dependencies right after we figure out the go.mod`.